x/wasm/keeper/msg_server_integration_test.go

```diff
@@ -141,3 +146,104 @@ func TestInstantiateContract(t *testing.T) {
 		})
 	}
 }
+
+func TestMigrateContract(t *testing.T) {
+	wasmApp := app.Setup(false)
+	ctx := wasmApp.BaseApp.NewContext(false, tmproto.Header{Time: time.Now()})
+
+	var (
+		myAddress       sdk.AccAddress = make([]byte, types.ContractAddrLen)
+		_, _, otherAddr                = testdata.KeyTestPubAddr()
+	)
+
+	specs := map[string]struct {
+		addr   string
+		expErr bool
+	}{
+		"admin can migrate a contract": {
+			addr:   myAddress.String(),
+			expErr: false,
+		},
+		"other address cannot migrate a contract": {
+			addr:   otherAddr.String(),
+			expErr: true,
+		},
+	}
+	for name, spec := range specs {
+		t.Run(name, func(t *testing.T) {
+			xCtx, _ := ctx.CacheContext()
+			// setup
+			_, _, sender := testdata.KeyTestPubAddr()
+			msg := types.MsgStoreCodeFixture(func(m *types.MsgStoreCode) {
+				m.WASMByteCode = hackatomContract
+				m.Sender = sender.String()
+			})
+
+			// store code
+			rsp, err := wasmApp.MsgServiceRouter().Handler(msg)(xCtx, msg)
+			require.NoError(t, err)
+			var storeCodeResponse types.MsgStoreCodeResponse
+			require.NoError(t, wasmApp.AppCodec().Unmarshal(rsp.Data, &storeCodeResponse))
+
+			// instantiate contract
+			initMsg := keeper.HackatomExampleInitMsg{
+				Verifier:    sender,
+				Beneficiary: myAddress,
+			}
+			initMsgBz, err := json.Marshal(initMsg)
+			require.NoError(t, err)
+
+			msgInstantiate := &types.MsgInstantiateContract{
+				Sender: sender.String(),
+				Admin:  myAddress.String(),
+				CodeID: storeCodeResponse.CodeID,
+				Label:  "test",
+				Msg:    initMsgBz,
+				Funds:  sdk.Coins{},
+			}
+			rsp, err = wasmApp.MsgServiceRouter().Handler(msgInstantiate)(xCtx, msgInstantiate)
+			require.NoError(t, err)
+			var instantiateResponse types.MsgInstantiateContractResponse
+			require.NoError(t, wasmApp.AppCodec().Unmarshal(rsp.Data, &instantiateResponse))
+			contract_addr := rsp.Data
```

Comment:
There seem to be some unexpected characters at the beginning of rsp.Data
```suggestion
			contract_addr := instantiateResponse.Address
```